common/dbus/platform/asio_libdbus/dbus.cpp

```diff
@@ -420,6 +432,23 @@ ExpectedStringPair ExtractValueFromDBusMessage(DBusMessage *message) {
 	return std::pair<string, string> {string(value1), string(value1)};
 }
 
+template <>
+expected::ExpectedBool ExtractValueFromDBusMessage(DBusMessage *message) {
+	DBusError dbus_error;
+	dbus_error_init(&dbus_error);
+	bool result;
+	if (!dbus_message_get_args(
+			message, &dbus_error, DBUS_TYPE_BOOLEAN, &result, DBUS_TYPE_INVALID)) {
+		auto err = MakeError(
+			ValueError,
+			string("Failed to extract reply data from reply message: ") + dbus_error.message + " ["
```

Comment:
Something needs to be casted so that the string's `+` operator kicks in. To me it makes most sense to do it at the first element of the expression and just append things to it with `+`.